cachi2/core/package_managers/yarn/main.py

```diff
@@ -73,6 +75,14 @@ def _configure_yarn_version(project: Project) -> None:
             ),
         )
 
+    # Note (mypy): version cannot be None anymore after the next statement
+    version = yarn_path_version if yarn_path_version else package_manager_version
+    if version < VersionInfo(3, 0, 0) or version >= VersionInfo(4, 0, 0):  # type: ignore
+        raise PackageRejected(
+            f"Unsupported Yarn version '{version}' detected",
+            solution="Please pick a valid release version of Yarn from the <3.0.0, 4.0.0) range",
+        )
```

Comment:
just a nitpick, I would stick to the mathematical conventions for intervals

e.g. "Please pick a supported Yarn release version within the interval **[**3.0.0, 4.0.0)"

Reply:
I might be the stupid one, but I'd have no clue what `[ )` means without the context, and would only assume it's a typo :laughing:

Reply:
> I think it's easy to assume 4.0.0 is included, but again, I suck bad at math 🤷‍♂️

fair enough, changed - how about now? :)